src/Extensions/SerializeExtensions.cs

```diff
@@ -66,10 +66,17 @@ public static string Serialize<T>(this T value)
             string result;
 
             XmlSerializer objectSerializer = new XmlSerializer(typeof(T));
-            
+
+            XmlWriterSettings settings = new XmlWriterSettings
+            {
+                Encoding = new UTF8Encoding(false),
+                OmitXmlDeclaration = true
+            };
+
             using (MemoryStream ms = new MemoryStream())
+            using (var writer = XmlWriter.Create(ms, settings))
             {
-                objectSerializer.Serialize(ms, value);
+                objectSerializer.Serialize(writer, value);
                 result = Encoding.UTF8.GetString(ms.ToArray());
```

Comment:
This is all a bit confusing (to me, at least), so we create an XmlWriter from a MemoryStream, and then write the input value to the XmlWriter but we were reading from the MemoryStream and returning the string. Hope that's the intent.

Reply:
It was to me too! So because of other changes via .NET6 we now are using the XmlWriter to feed the MemoryStream. This is done to eliminate the inclusion (again .NET6) of a BOM (byte order mark) as well as the XmlDeclaration (which now enforces "encoding=blabla" which was causing another error in the database.

So yes, long story short, we're using the XmlWriter with the correct settings to feed the memory stream to get the correct xml string that is being asked for.